src/extension.js

```diff
@@ -102,10 +102,19 @@ function activate(context) {
             return;
         }
 
-        // the function isFileNameOk checks for the include and exclude settings
+        // check if this file/folder isn't on the exclude settings list
         if (!util.isFileNameOk(activeEditor.document.fileName)) {
             return;
         }
+        // check if we have a list of languages to be used for. If not: use on all languages.
+        const thisFileLangId = activeEditor.document.languageId;
+        console.log(`thisFileLangId: ${thisFileLangId}`);
+        const includedLanguages = settings.get('includedLanguages') || [];
+        console.log(`includedLanguages: ${String(includedLanguages)} (${typeof includedLanguages})`);
+        if (includedLanguages && includedLanguages.length > 0 && !includedLanguages.includes(thisFileLangId)) {
+            console.log(`updateDecorations not wanted on lang ${thisFileLangId} for file ${activeEditor.document.fileName}`);
+            return;
+        }
```

Comment:
Why is this not in the function that checks if the current file should be used?

Reply:
There was a reason at the time, but I now forget why.
But it may not be relevant now, depending how the discussion on which type of `include` setting to use.